### PR TITLE
Fix source semantics of shifts

### PR DESCRIPTION
**Pull request by vbgl**
This depends on:
  - [x] #412
  - [x] https://github.com/formosa-crypto/libjade/pull/59

### CODE REVIEW

**Review by bgregoir — APPROVED** [review]
I think we should wait for the merge of the two related PR before merging this one.
Is it right?

**Comment by vbgl** [comment]
The second commit must not be merged. It is just here to show that the proposed changes to `libjade` are enough.

**Pull request merged**
